internal/interface/grpc/handlers/arkservice.go

```diff
@@ -258,6 +270,62 @@ func (h *handler) GetEventStream(
 	}
 }
 
+func (h *handler) UpdateStreamTopics(
+	ctx context.Context,
+	req *arkv1.UpdateStreamTopicsRequest,
+) (*arkv1.UpdateStreamTopicsResponse, error) {
+	if req.GetStreamId() == "" {
+		return nil, status.Error(codes.InvalidArgument, "missing stream id")
+	}
+
+	switch topicsChange := req.GetTopicsChange().(type) {
+	case nil:
+		return nil, status.Error(codes.InvalidArgument, "no topics provided")
+	// when overwrite topics is provided, it takes precedence, we will not
+	// process add/remove topics in this case
+	case *arkv1.UpdateStreamTopicsRequest_Overwrite:
+		if topicsChange.Overwrite == nil {
+			return nil, status.Error(codes.InvalidArgument, "overwrite topics is nil")
+		}
```

Comment:
a long time ago, we made the decision to use only getters, please fix like so also in the rest of the switch:

```suggestion
		if req.GetOverwrite() == nil {
			return nil, status.Error(codes.InvalidArgument, "missing topics to overwrite")
		}
```